note.go

```diff
@@ -0,0 +1,136 @@
+package hubspot
+
+const (
+	noteBasePath = "notes"
+)
+
+// NoteService is an interface of note endpoints of the HubSpot API.
+// HubSpot notes store information about interactions and communications.
+// It can be associated with other CRM objects such as contact, company, and deal.
+// Reference: https://developers.hubspot.com/docs/api/crm/notes
+type NoteService interface {
+	Get(noteID string, note interface{}, option *RequestQueryOption) (*ResponseResource, error)
+	Create(note interface{}) (*ResponseResource, error)
+	Update(noteID string, note interface{}) (*ResponseResource, error)
+	Delete(noteID string) error
+	AssociateAnotherObj(noteID string, conf *AssociationConfig) (*ResponseResource, error)
+}
+
+// NoteServiceOp handles communication with the note related methods of the HubSpot API.
+type NoteServiceOp struct {
+	notePath string
+	client   *Client
+}
+
+var _ NoteService = (*NoteServiceOp)(nil)
+
+// Note represents a note in HubSpot.
+type Note struct {
```

Comment:
I checked the documentation but couldn't find any mention of `hs_note_title`.
I may have missed something, but if you have any references please let me know.

Reply:
Thanks, you are correct this is not a valid field. I removed it.